cmp/grid/impl/ColumnWidthCalculator.js

```diff
@@ -87,26 +84,38 @@ export class ColumnWidthCalculator {
         } catch (e) {
             console.warn(`Error calculating max data width for column "${column.colId}".`, e);
         } finally {
-            this.setCellElActive(false);
+            this.resetClassNames();
         }
     }
 
     calcLevelWidth(gridModel, records, column, options, indentationPx = 0) {
         const {field, getValueFn, renderer} = column,
+            {sizingMode} = gridModel,
             {bufferPx} = options;
 
         // 1) Get unique values
-        const values = new Set();
+        const values = new Set(),
+            valueClassMap = {};
+
         records.forEach(record => {
             if (!record) return;
             const ctx = {record, field, column, gridModel, store: gridModel.store},
                 rawValue = getValueFn(ctx),
-                value = renderer ? renderer(rawValue, ctx) : rawValue;
+                value = renderer ? renderer(rawValue, ctx) : rawValue,
+                cellClass = this.getCellClass(gridModel, column, record, rawValue),
+                rowClass = this.getRowClass(gridModel, record);
+
             values.add(value);
+
+            // 1b) Track combinations of cell and row classes applied to this value.
+            // Stringify the combination to test Set equality.
+            let classNames = valueClassMap[value];
+            if (!classNames) classNames = valueClassMap[value] = new Set();
+            classNames.add(JSON.stringify({cellClass, rowClass}));
```

Comment:
Tom and I discussed doing this work of looking up the cells and row classes only after we have identified the records with the longest values.

Also, why not uniquify these things as a simple string with a pipe and skip the jsonification?

Reply:
Good suggestion about uniquifiying with a string pipe, will make that change